### PR TITLE
Bug fix

### PR DESCRIPTION
**Pull request by marvinfriede**
- fix package_data in setup.cfg (v0.1.1 does not package non-Python data, omitting the npy file!)
- the C6 coefficients show minor deviations because of floating point inaccuracies and small values
- improve typing to fix mypy errors
- rigorous gradient tests (includes some `nan` fixes)
- hessian test

### CODE REVIEW

File: src/tad_dftd3/util.py [hunk 1]
```diff
@@ -44,6 +44,55 @@ def real_triples(numbers: Tensor, diagonal: bool = False) -> Tensor:
...
+
+    # unsqueeze different dimension to create matrix
+    differences = x.unsqueeze(-2) - y.unsqueeze(-3)
+
+    distances = torch.sum(torch.pow(differences, p), -1)
+    return torch.sqrt(torch.clamp(distances, min=eps))
+
+
+def cdist(
+    x: Tensor, y: Optional[Tensor] = None, p: int = 2, **_: dict[str, Any]
```
Comment (awvwgk) [inline src/tad_dftd3/util.py:91]: Why are additional arguments dropped here?

Comment (marvinfriede) [reply src/tad_dftd3/util.py:91]: I am currently slightly reworking this part, but I was hoping that this is just a temporary replacement, and we can use `torch.cdist` in the future again. So I did not want to replace all `torch.cdist` calls, which currently look like this:

```python
torch.cdist(positions, positions, p=2, compute_mode="use_mm_for_euclid_dist" )
```

But then, I do not think it is likely that PyTorch would include handling of zero division for the backward of `torch.cdist`. So I think we should switch to one of the manual implementations.

Comment (marvinfriede) [reply src/tad_dftd3/util.py:91]: I just realized that we could do something like:

```python
def cdist(x: Tensor, y: Optional[Tensor] = None, p: int = 2) -> Tensor:
  return torch.cdist(x, y, p=p, compute_mode="use_mm_for_euclid_dist")
```

if we wanted to switch back. So I will remove the unnecessary arguments.

Comment (awvwgk) [reply src/tad_dftd3/util.py]: Maybe habe a look into `functools.partial` here, which might be clearer than creating thin wrapper functions

Comment (marvinfriede) [reply src/tad_dftd3/util.py]: I don't quite see, how this could make clearer. We will not use `torch.cdist` anymore, only `cdist`. And in the end, one should only call this wrapper.

Comment (marvinfriede) [reply src/tad_dftd3/util.py]: @awvwgk Do you still consider this necessary? Otherwise, I would like to merge and fix the release (the current one is broken because of the missing npy file).

**Review by awvwgk — APPROVED** [review]
